replit_river/server_transport.py

```diff
@@ -148,30 +146,74 @@ async def _establish_handshake(
                 websocket,
             )
             raise InvalidMessageException("handshake request to wrong server")
-        if handshake_request.expectedSessionState is None:
-            # TODO: remove once we have upgraded all clients
-            my_session_id = await self._get_or_create_session_id(
-                request_message.from_, handshake_request.sessionId
+
+        old_session = self._sessions.get(request_message.from_, None)
```

Comment:
Not sure if we could run into a race condition here, case might be :
1. we got a old_session here
2. control was given out while doing `await old_session.get_next_sent_seq()`
3. old session was deleted
4. When the control is given back the session is already gone

We probably want to wrap these logic with the session lock

Reply:
which lock do we acquire?

Reply:
is it sufficient to wrap with `async with old_session._ws_lock, old_session._state_lock:`